cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/DistributedMrJobInfoFetcher.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.mapreduce;
+
+import co.cask.cdap.common.exception.NotFoundException;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.MRJobInfo;
+import com.google.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+
+/**
+ * Retrieves information about a run of a MapReduce job, using {@link MRJobClient} and
+ * then {@link LocalMRJobInfoFetcher} if necessary.
+ */
+public class DistributedMRJobInfoFetcher implements MRJobInfoFetcher {
```

Comment:
Seems like the file name doesn't match with class name. (Mr in File name). you have to turn on the git case aware.